deserialization.go

```diff
@@ -94,7 +94,7 @@ func read[B any](context context.Context, dec decoder) (B, error) {
 	var body B
 
 	err := dec.Decode(&body)
-	if err != nil {
+	if err != nil && err != io.EOF {
```

Comment:
I would expect this.

```suggestion
	if err != nil && !errors.Is(err, io.EOF) {
```

Reply:
Good call, thanks